ci/do_ci.sh

```diff
@@ -282,15 +292,17 @@ elif [[ "$1" == "bazel.legacy.test" ]]; then
   exit 0
 elif [[ "$1" == "bazel.noexcept" ]]; then
   # there are some exceptions and error handling code from the Prometheus and Jaeger Clients
+  # as well as Opentracing shim (due to some third party code in its Opentracing dependency)
   # that make this test always fail. ignore Prometheus and Jaeger exporters in the noexcept here.
-  bazel $BAZEL_STARTUP_OPTIONS build --copt=-fno-exceptions --build_tag_filters=-jaeger $BAZEL_OPTIONS_ASYNC -- //... -//exporters/prometheus/... -//exporters/jaeger/... -//examples/prometheus/... -//sdk/test/metrics:attributes_hashmap_test
-  bazel $BAZEL_STARTUP_OPTIONS test --copt=-fno-exceptions --build_tag_filters=-jaeger $BAZEL_TEST_OPTIONS_ASYNC -- //... -//exporters/prometheus/... -//exporters/jaeger/... -//examples/prometheus/... -//sdk/test/metrics:attributes_hashmap_test
+  bazel $BAZEL_STARTUP_OPTIONS build --copt=-fno-exceptions --build_tag_filters=-jaeger $BAZEL_OPTIONS_ASYNC -- //... -//exporters/prometheus/... -//exporters/jaeger/... -//examples/prometheus/... -//sdk/test/metrics:attributes_hashmap_test -//opentracing-shim/...
```

Comment:
nit - Just wondering if we can use `dynamic_cast` if OPENTELEMETRY_RTTI_ENABLED is enabled, and use `static_cast` otherwise ?

Reply:
One inconvenience with static_cast is that I do not know if the downcasting succeeds, so I cannot simply assume that I have a SpanContextShim as this would be undefined behavior. My idea is using a 64bit tag to disambiguate instances of the shim class which will make it a bit safer to call `context()` and `baggage()`, both of which are methods specific to shim only.